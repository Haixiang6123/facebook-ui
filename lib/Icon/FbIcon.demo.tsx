import * as React from 'react'
import FbExample from './FbIcon.example'
import Demo from '../../Demo'

export default () => {
  return (
    <Demo code={require('!!raw-loader!./FbIcon.example.tsx').default}>
      <FbExample/>
    </Demo>
  )
}
