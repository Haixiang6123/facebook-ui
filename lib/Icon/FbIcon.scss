@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.fb-icon {
  width: 1.4em;
  height: 1.4em;
  fill: currentColor;
  &.fb-icon-spin {
    animation: spin .7s infinite linear;
  }
}
