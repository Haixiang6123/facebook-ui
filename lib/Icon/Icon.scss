.fb-icon {
  width: 1em;
  height: 1em;
  fill: #4267b2;
}
