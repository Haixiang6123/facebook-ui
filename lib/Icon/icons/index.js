let exportAllIcons = (requireContext) => requireContext.keys().forEach(requireContext)

try {
    exportAllIcons(require.context('./', true, /\.svg$/))
} catch (error) {
}