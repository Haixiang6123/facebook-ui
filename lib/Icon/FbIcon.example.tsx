import * as React from 'react'
import FbIcon from './FbIcon'

export default function () {
  return (
    <div>
      <FbIcon name="video"/>
      <FbIcon name="sport"/>
    </div>
  )
}
