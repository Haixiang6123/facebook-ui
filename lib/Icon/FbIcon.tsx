import * as React from 'react'
import classes from '../utils/classes'
import './icons/index'
import './FbIcon.scss'

interface IconProps extends React.SVGAttributes<SVGElement> {
  name: string
  solid?: boolean
  spin?: boolean
}

const FbIcon: React.FunctionComponent<IconProps> =
  ({className, name, solid, spin, ...restProps}) => {
    const link = `#icon_${name}` + (solid ? '_fill' : '')
    const classNames = classes('fb-icon', className, spin && 'fb-icon-spin')
    return (
      <svg className={classNames} {...restProps}>
        <use xlinkHref={link}/>
      </svg>
    )
}

export default FbIcon
