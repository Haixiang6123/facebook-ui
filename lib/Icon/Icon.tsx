import * as React from 'react'
import classes from '../utils/classes'
import './icons/index'
import './Icon.scss'

interface IconProps extends React.SVGAttributes<SVGElement> {
  name: string
}

const Icon: React.FunctionComponent<IconProps> = ({className, name, ...restProps}) => {
  return (
    <svg className={classes('fb-icon', className)} {...restProps}>
      <use xlinkHref={`#icon_${name}`}/>
    </svg>
  )
}

export default Icon
