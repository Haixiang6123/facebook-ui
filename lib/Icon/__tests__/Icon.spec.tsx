import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import Icon from '../Icon'

describe('Icon', () => {
  it('can render', () => {
    const json = renderer.create(<Icon name="video"/>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('can handle event', () => {
    const handler = jest.fn()
    const component = mount(<Icon name="video" onClick={handler}/>)
    component.find('svg').simulate('click')
    expect(handler).toBeCalled()
  })
})
