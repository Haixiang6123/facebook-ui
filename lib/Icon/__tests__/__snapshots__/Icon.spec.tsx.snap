// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon can render 1`] = `
<svg
  className="fb-icon"
>
  <use
    xlinkHref="#icon_video"
  />
</svg>
`;
